share/wake/lib/system/io.wake

```diff
@@ -15,6 +15,40 @@
 
 package wake
 
+export data PathType =
+    PathTypeRegularFile
+    PathTypeDirectory
+    PathTypeCharacterDevice
+    PathTypeBlockDevice
+    PathTypeFifo
+    PathTypeSymlink
+    PathTypeSocket
+
+export tuple Stat =
+    Type: PathType
+    Mode: Integer
```

Comment:
Can we/should we provide a better wrapper around mode? Just because linux uses a int here doesn't mean we have to. 

I could imagine something like 

```
export tuple Permission =
    Read: Bool
    Write: Bool
    Execute: Bool

export def Mode = 
    User: Permission
    Group: Permission
    Other: Permission
```

Reply:
I thought the same thing but I got lazy and didn't do it. I'll work on it.